packages/js/src/plugins/auctionHouseModule/executeSale.ts

```diff
@@ -0,0 +1,155 @@
+import { ConfirmOptions, PublicKey } from '@solana/web3.js';
+import type { Metaplex } from '@/Metaplex';
+import { TransactionBuilder } from '@/utils';
+import { createExecuteSaleInstruction } from '@metaplex-foundation/mpl-auction-house';
+import {
+  useOperation,
+  Operation,
+  OperationHandler,
+  Signer,
+  toPublicKey,
+  token,
+  lamports,
+  amount,
+  SolAmount,
+  SplTokenAmount,
+} from '@/types';
+import { SendAndConfirmTransactionResponse } from '../rpcModule';
+import { findAssociatedTokenAccountPda } from '../tokenModule';
+import { findMetadataPda } from '../nftModule';
+import { AuctionHouse } from './AuctionHouse';
+import {
+  findAuctionHouseBuyerEscrowPda,
+  findAuctionHouseProgramAsSignerPda,
+  findAuctionHouseTradeStatePda,
+} from './pdas';
+
+// -----------------
+// Operation
+// -----------------
+
+const Key = 'ExecuteSaleOperation' as const;
+export const executeSaleOperation = useOperation<ExecuteSaleOperation>(Key);
+export type ExecuteSaleOperation = Operation<
+  typeof Key,
+  ExecuteSaleInput,
+  ExecuteSaleOutput
+>;
+
+export type ExecuteSaleInput = {
+  auctionHouse: AuctionHouse;
+  buyer: PublicKey | Signer;
+  seller?: PublicKey | Signer; // Default: identity
+  mintAccount: PublicKey; // Required for checking Metadata
+  sellerTradeState: PublicKey;
+  buyerTradeState: PublicKey;
+  price: SolAmount | SplTokenAmount;
+  tokens?: SplTokenAmount; // Default: token(1)
```

Comment:
That's not really how I had it in my head to be honest.

Since we creating a listing or a bid we can the `Listing` or `Bid` models respectively, I was thinking we could just pass the two there with the `AuctionHouse` model and we should have all the information we need to execute the sale. We could even make a quick check to ensure that the data on the listing and the bid match before sending the transaction and throw a custom error otherwise.

What do you think?